### PR TITLE
Unit test for rule & codec & disconnect tcp & broker_tcp v5

### PR DESCRIPTION
**Pull request by Hermann0222**
fixes #<issue number> <issue synopsis>

<Comments describing your change. Not all changes need this.>

Note that the above format should be used in your git commit comments.
You agree that by submitting a PR, you have read and agreed to our
contributing guidelines.

### CODE REVIEW

File: include/nng/mqtt/mqtt_client.h [hunk 1]
```diff
@@ -481,6 +481,8 @@ NNG_DECL void      nng_mqtt_msg_set_unsubscribe_property(nng_msg *, property *);
 NNG_DECL void nng_mqtt_msg_set_unsuback_return_codes(
     nng_msg *, uint8_t *, uint32_t);
 NNG_DECL uint8_t *nng_mqtt_msg_get_unsuback_return_codes(nng_msg *, uint32_t *);
+NNG_DECL property *nng_mqtt_msg_get_unsuback_property(nng_msg *);
+NNG_DECL void      nng_mqtt_msg_set_unsuback_property(nng_msg *, property *);
```
Comment (JaylinYu) [inline include/nng/mqtt/mqtt_client.h:485]: missing before?

File: src/supplemental/mqtt/mqtt_msg.c [hunk 2]
```diff
@@ -508,7 +509,7 @@ nni_mqtt_msg_set_unsubscribe_topics(
 	for (size_t i = 0; i < topic_count; i++) {
 		nni_mqtt_topic_array_set(
 		    proto_data->payload.unsubscribe.topic_arr, i,
-		    (const char *) topics[i].buf);
+		    (const char *) topics[i].buf, topics[i].length);
```
Comment (JaylinYu) [inline src/supplemental/mqtt/mqtt_msg.c:512]: previous bug?

**Review by JaylinYu — APPROVED** [review]
Great work

**Review by lee-emqx — APPROVED** [review]
Good job!

File: include/nng/mqtt/mqtt_client.h (hunk 1 above)
Comment (Hermann0222) [reply include/nng/mqtt/mqtt_client.h:485]: yes

File: src/supplemental/mqtt/mqtt_msg.c (hunk 2 above)
Comment (Hermann0222) [reply src/supplemental/mqtt/mqtt_msg.c:512]: I think so. Length is an unused arg in unsub before, now it is fixed and works just like sub.

**Pull request merged**